requirements-dev.txt

```diff
@@ -1,7 +1,6 @@
 black~=22.0
 stestr
-astroid==2.5
```

Comment:
The pylint and astroid versions are always tightly coupled in my experience.

The options are:

1. Only specify pylint version
2. Specify both astroid and pylint

The downside to 1 is that someone's astroid/pylint versions gets out of sync (I think pip is better about avoiding / warning about this than it used to be). The downside to 2 is that someone needs to remember to update both versions which might be a little confusing but the CI wouldn't let them forget to do this.

Since we are already pinning the pylint version, maybe option 2 has less potential downsides.